File: common/data_loader.rs
use std::fs;
use std::path::Path;
use serde::Deserialize;

#[derive(Deserialize)]
struct Record {
    payload: String,
}

pub fn load_data(path: &str) -> Vec<String> {
    let mut data = Vec::new();
    let p = Path::new(path);

    if p.is_dir() {
        for entry in fs::read_dir(p).unwrap() {
            let entry = entry.unwrap();
            let content = fs::read_to_string(entry.path()).unwrap();
            data.push(content);
        }
    } else if path.ends_with(".jsonl") {
        let content = fs::read_to_string(path).unwrap();
        for line in content.lines() {
            let record: Record = serde_json::from_str(line).unwrap();
            data.push(record.payload);
        }
    } else if path.ends_with(".csv") {
        let content = fs::read_to_string(path).unwrap();
        for line in content.lines().skip(1) { // skip header
            data.push(line.to_string());
        }
    }
    data
}

File: common/data_loader.py
import json
import csv
import os

def load_data(path):
    data = []
    if os.path.isdir(path):
        for filename in os.listdir(path):
            with open(os.path.join(path, filename), 'r') as f:
                data.append(f.read())
    elif path.endswith('.jsonl'):
        with open(path, 'r') as f:
            for line in f:
                data.append(json.loads(line)['payload'])
    elif path.endswith('.csv'):
        with open(path, 'r') as f:
            reader = csv.reader(f)
            next(reader) # skip header
            for row in reader:
                data.append(row[0])
    return data
